### PR TITLE
User can add a playlist with /playlists endpoint

### LINKED ISSUE #26: Happy Path: User can add a playlist with /playlists endpoint
User sends **UNIQUE** title in request body to `POST /api/v1/playlists`

User can expect a successful response body as follows with a `201 status`:
```
{
  "id": 1,
  "title": "Cleaning House",
  "createdAt": 2019-11-26T16:03:43+00:00,
  "updatedAt": 2019-11-26T16:03:43+00:00,
}
```

Acceptance criteria:
- [x]  All tests passing
- [x]  Happy path
- [ ]  Peer review
- [ ]  Deployed to Heroku

### PR DESCRIPTION
**Pull request by zacisaacson**
## Type of change 

- [x] :hatching_chick: New feature (non-breaking change which adds functionality)
- [x] :page_with_curl: This change requires a documentation update

## Summary of changes 

- Create happy path spec
- Create sad path specs for not sending title and unique title
- Complete functionality

## How Has This Been Tested?

- [x] :black_square_button: Integration testing 

## Checklist:

- [x] :white_check_mark: I have performed a self-review of my own code
- [x] :eight_spoked_asterisk: I have commented my code, particularly in hard-to-understand areas
- [x] :no_entry_sign: My changes generate no new warnings
- [x] :heavy_check_mark: I have added tests that prove my fix is effective or that my feature works
- [x] :100: New and existing unit tests pass locally with my changes

### CODE REVIEW

File: routes/api/v1/playlists.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const fetch = require('node-fetch');
+const environment = process.env.NODE_ENV || 'development';
+const configuration = require('../../../knexfile')[environment];
+const database = require('knex')(configuration);
+
+router.post('/', (request, response) => {
+  if (request.body.title) {
+    database('playlists').insert({title: request.body.title}, 'id')
+    .then(playlist => {
+      database('playlists').where('id', playlist[0])
```
Comment (zacisaacson) [inline routes/api/v1/playlists.js]: Interested to see if we can implement this without another DB call

File: tests/playlists/addPlaylist.spec.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      const res = await request(app)
+        .post('/api/v1/playlists')
+        .send({
+          title: "Roadtrip!!!",
+        });
+
+      expect(res.statusCode).toBe(400);
+      expect(res.body).toHaveProperty('error')
+      expect(res.body.error).toBe("Title must be unique!");
+    });
```
Comment (zacisaacson) [inline tests/playlists/addPlaylist.spec.js]: Thoughts on any missing sad paths?

File: routes/api/v1/playlists.js (hunk 1 above)
Comment (lrs8810) [reply routes/api/v1/playlists.js]: Not sure at this time, but we can make a card to refactor.

File: tests/playlists/addPlaylist.spec.js (hunk 2 above)
Comment (lrs8810) [reply tests/playlists/addPlaylist.spec.js]: These look great!

**Pull request merged**